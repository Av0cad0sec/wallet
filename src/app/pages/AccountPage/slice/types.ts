import { AccountsRow, OperationsRow } from 'vendors/explorer'

export interface BalanceDetails {
  total: number
  debonding: number
  escrow: number
  available: number
  delegations: number
}

/* --- STATE --- */
export interface AccountState extends AccountsRow {
  loading: boolean
  transactions: OperationsRow[]
}
