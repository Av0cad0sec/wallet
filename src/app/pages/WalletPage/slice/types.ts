import { ErrorPayload } from 'types/errors'

export enum TransactionType {
  SENT = 0,
  RECEIVED = 1,
}

export enum WalletType {
  Ledger = 'ledger',
  PrivateKey = 'private_key',
  Mnemonic = 'mnemonic',
}

export interface Transaction {
  address: string
  date: number
  amount: number
  type: TransactionType
}

export interface TransactionSent {
  from: string
  to: string
  amount: number
}
export interface SendTransactionPayload {
  /* bech32 Address */
  to: string

  /* Token amount */
  amount: number
}

/**
 * WalletBalance
 *
 * We stock the amounts as strings to work around
 * poor bigint support in JS
 *
 */
export interface WalletBalance {
  available: string
  escrow: string
  debonding: string
}

export interface Wallet {
  publicKey: string
  address: string
  type?: WalletType
  privateKey?: string
}

export interface TransactionStatus {
  success: boolean
  error?: ErrorPayload
}

/* --- STATE --- */
export interface WalletState extends Wallet {
  isOpen: boolean
  balance: WalletBalance
  transaction: TransactionStatus
}
