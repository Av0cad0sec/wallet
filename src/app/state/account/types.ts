import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'

export interface BalanceDetails {
  available: number | null
  debonding: number | null
  delegations: number | null
  total: number | null
}

export interface Account extends BalanceDetails {
  address: string
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
