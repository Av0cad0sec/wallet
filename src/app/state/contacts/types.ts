export interface Contact {
  address: string
  name: string
}

/* --- STATE --- */
export interface ContactsState {
  [address: string]: Contact
}
