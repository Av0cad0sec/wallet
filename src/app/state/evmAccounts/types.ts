export interface EvmAccount {
  ethPrivateKey: string
  ethAddress: string
}

/* --- STATE --- */
export interface EvmAccounts {
  [ethAddress: string]: EvmAccount
}
