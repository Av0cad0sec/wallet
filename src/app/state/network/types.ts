/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string
}
