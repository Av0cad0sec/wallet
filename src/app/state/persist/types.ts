import { RootState } from 'types'

type StrictOmit<T, K extends keyof T> = Omit<T, K>

interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface RootStateWithoutPersistSlice extends StrictOmit<RootState, 'persist'> {
  persist: null
}

export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
