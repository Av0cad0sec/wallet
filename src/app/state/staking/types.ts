/* --- STATE --- */
export interface Validator {
  name?: string
  address: string
  escrow?: number
  status: string
  rank: number
  fee: number
  media?: ValidatorMediaInfo
}

export interface CommissionBounds {
  lower: number
  upper: number
  epochStart: number
  epochEnd?: number
}

export interface ValidatorDetails {
  currentCommissionBounds?: CommissionBounds
  scheduledCommissionBounds?: CommissionBounds[]
  nextCommission?: {
    fee: number
    epochStart: number
  }
}
export interface ValidatorMediaInfo {
  website_link?: string
  email_address?: string
  twitter_acc?: string
  tg_chat?: string
}

export interface Delegation {
  validator?: Validator
  amount: string
  shares: string
  validatorAddress: string
}

export interface DebondingDelegation extends Delegation {
  epoch: number
}

export interface StakingState {
  /** List of all the validators */
  validators: Validator[]

  /** List of active delegations for the selected account */
  delegations: Delegation[]

  /** List of debonding delegations for the selected account */
  debondingDelegations: DebondingDelegation[]

  /** Addresss of the selected validator */
  selectedValidator?: string | null

  /** Lazy-loaded validator details (commission bounds and rates schedules) */
  selectedValidatorDetails: ValidatorDetails | null

  loading: boolean
}
