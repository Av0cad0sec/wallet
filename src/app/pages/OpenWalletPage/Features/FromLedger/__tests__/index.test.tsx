import { cleanup, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ledgerActions } from 'app/state/ledger'
import { LedgerStep } from 'app/state/ledger/types'
import { walletActions } from 'app/state/wallet'
import * as React from 'react'
import { Provider, useDispatch } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { FromLedger } from '..'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
  useDispatch: jest.fn(),
}))

const renderComponent = (store, abortFunction = () => {}) =>
  render(
    <Provider store={store}>
      <FromLedger abort={abortFunction} />
    </Provider>,
  )

describe('<FromLedger  />', () => {
  let store: ReturnType<typeof configureAppStore>
  let mockDispatch = useDispatch as jest.Mock<typeof useDispatch>

  beforeEach(() => {
    store = configureAppStore()
    mockDispatch.mockImplementation(() => jest.fn())
  })

  afterEach(() => {
    jest.resetModules()
    jest.resetAllMocks()
    cleanup()
    document.body.innerHTML = ''
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component).toMatchSnapshot()
    screen.getByTestId('ledger-open-accounts')
  })

  it('should list the accounts when done', () => {
    const component = renderComponent(store)
    store.dispatch(
      ledgerActions.accountsListed([
        {
          address: 'oasis1abcdefghijkl',
          balance: { available: '0', debonding: '0', escrow: '0', total: '0' },
          path: [44, 474, 0],
          publicKey: '00',
          selected: false,
        },
      ]),
    )

    store.dispatch(ledgerActions.setStep(LedgerStep.Done))
    expect(component.getByText('oasis1 abcd efgh ijkl')).toBeInTheDocument()
  })

  it('should open the selected accounts', () => {
    const dispatchFn = jest.fn()
    mockDispatch.mockImplementation(() => dispatchFn)

    const component = renderComponent(store)
    store.dispatch(
      ledgerActions.accountsListed([
        {
          address: 'oasis1aaaaaaaa',
          balance: { available: '0', debonding: '0', escrow: '0', total: '0' },
          path: [44, 474, 0],
          publicKey: '00',
          selected: false,
        },
        {
          address: 'oasis1bbbbbbbb',
          balance: { available: '0', debonding: '0', escrow: '0', total: '0' },
          path: [44, 474, 1],
          publicKey: '00',
          selected: false,
        },
      ]),
    )

    store.dispatch(ledgerActions.setStep(LedgerStep.Done))
    userEvent.click(screen.getByText('oasis1 aaaa aaaa'))
    expect(dispatchFn).toHaveBeenLastCalledWith({ payload: 0, type: ledgerActions.toggleAccount.type })
    store.dispatch(ledgerActions.toggleAccount(0))

    userEvent.click(screen.getByTestId('ledger-open-accounts'))
    expect(dispatchFn).toHaveBeenLastCalledWith(
      expect.objectContaining({ type: walletActions.openWalletsFromLedger.type }),
    )
  })
})
