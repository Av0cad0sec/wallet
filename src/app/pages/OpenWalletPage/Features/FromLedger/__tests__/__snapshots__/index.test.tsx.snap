// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FromLedger  /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 750px;
  padding: 24px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-top: 48px;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c7 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c5 {
  margin-top: 0px;
  margin-bottom: 24px;
  font-size: 1;
  line-height: normal;
  max-width: 1200px;
  font-weight: 600;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: transparent;
  position: relative;
  z-index: 20;
  pointer-events: none;
  outline: none;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
}

.c1 {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  background-color: rgba(0,0,0,0.5);
  color: #f8f8f8;
  pointer-events: all;
  will-change: transform;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 48px;
  background-color: #FFFFFF;
  color: #444444;
  outline: none;
  pointer-events: all;
  z-index: 20;
  position: absolute;
  max-height: calc(100% - 0px - 0px);
  max-width: calc(100% - 0px - 0px);
  border-radius: 4px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-animation: hWAzxx 0.2s ease-in-out forwards;
  animation: hWAzxx 0.2s ease-in-out forwards;
}

.c3 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding-top: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    position: fixed;
    width: 100%;
    height: 100%;
    min-height: 100vh;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    position: relative;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    position: relative;
    max-height: none;
    max-width: none;
    border-radius: 0;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    -webkit-animation: none;
    animation: none;
    height: 100vh;
    width: 100vw;
  }
}

<body
    style="overflow: hidden;"
  >
    <div />
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="c0"
          tabindex="-1"
        >
          <div
            class="c1"
          />
          <div
            class="c2"
            data-g-portal-id="0"
          >
            <a
              aria-hidden="true"
              class="c3"
              tabindex="-1"
            />
            <div
              class="c4"
            >
              <h1
                class="c5"
                size="1"
              >
                Select the wallets to open
              </h1>
              <div
                class="c6"
              >
                <button
                  class="c7"
                  disabled=""
                  style="border-radius: 4px;"
                  type="button"
                >
                  Cancel
                </button>
                <div
                  class="c8"
                />
                <button
                  class="c9"
                  data-testid="ledger-open-accounts"
                  disabled=""
                  style="border-radius: 4px;"
                  type="button"
                >
                  Open
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
